modules/150-user-authn/images/basic-auth-proxy/app/pkg/proxy/provider/oidc.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"golang.org/x/oauth2"
+)
+
+type OIDCProvider struct {
+	apiURL   string
+	login    string
+	password string
+
+	allowedGroups map[string]struct{}
+
+	httpClient   *http.Client
+	oidcProvider *oidc.Provider
+	oauth2Config *oauth2.Config
+}
+
+func NewOIDCProvider(apiURL, login, password string, scopes, allowedGroups []string) *OIDCProvider {
+	client := &http.Client{
+		Timeout: 30 * time.Second,
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).DialContext,
+			MaxIdleConns:          10,
+			IdleConnTimeout:       30 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+			TLSClientConfig:       &tls.Config{InsecureSkipVerify: true},
```

Comment:
Isn't `InsecureSkipVerify` potentially risky. Looks possible to have MITM here. Is it for custom CA?

Reply:
This is a copy paste from the crowd provider. Probably, it is best to take the CA option from the provider, but not in the first round of implementation.